lib/config-initializer.js

```diff
@@ -128,11 +129,87 @@ function promptUser(callback) {
             name: "format",
             message: "What format do you want your config file to be in?",
             default: "JSON",
-            choices: ["JSON", "YAML"]
+            choices: ["JSON", "YAML"],
+            when: function(answers) {
```

Comment:
I think you're right, @nzakas.  The `when:` method should be on the jsx question.  Meaning, if someone uses React, then ask if they use JSX.